src/components/dockerRegistry/Docker.tsx

```diff
@@ -204,6 +206,8 @@ function DockerForm({
     });
     const [deleting, setDeleting] = useState(false);
     const [confirmation, toggleConfirmation] = useState(false);
+    const [isIAMAuthType, setIAMAuthType] = useState(!awsAccessKeyId && !awsSecretAccessKey)
```

Comment:
registryType check?

Reply:
It's done at the bottom where we're using this state.